py/server/deephaven/experimental/s3.py

```diff
@@ -84,6 +89,28 @@ def __init__(self,
             num_concurrent_write_parts (int): the maximum number of parts that can be uploaded concurrently when writing
                 to S3 without blocking, defaults to 64. Setting a higher value may increase throughput, but may also
                 increase memory usage.
+            profile_name (str): the default profile name used for configuring the default region, credentials, etc.,
+                when reading or writing to S3. If not provided, the AWS SDK picks the profile name from the
+                'aws.profile' system property, the "AWS_PROFILE" environment variable, or defaults to "default".
+                Setting a profile name assumes that the credentials are provided via this profile; if that is not the
+                case, you must explicitly set credentials.
+            config_file_path (str): the path to the configuration file to use for configuring the default region,
+                credentials, etc. when reading or writing to S3. If not provided, the AWS SDK picks the configuration
+                file from the 'aws.configFile' system property, the "AWS_CONFIG_FILE" environment variable, or defaults
+                to "{user.home}/.aws/config".
+                Setting a configuration file path assumes that the credentials are provided via the config and
+                credentials files; if that is not the case, you must explicitly set credentials.
```

Comment:
1. "assumes that the credentials are provided via the config and credentials files" is confusing.  Are credentials split up between these two files?  Can they be in either file?  etc.
2. Another spot where setting the credentials explicitly is not clear.

Reply:
Your confusion is very valid, and actually something AWS SDK is also not very clear on. 
Internally, both these files are combined, so yes, we can keep credentials, etc. in any of these files.
But there are conventions around what goes in what file, which can be seen in the examples here:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html

I have tried to make the language more clear and have linked the above page in the comment.
Let me know if its more clear or if you have any suggestions.